src/hls-adapter.js

```diff
@@ -972,12 +1074,14 @@ export default class HlsAdapter extends BaseMediaSourceAdapter {
         this.destroy();
       }
     } else {
-      const {category, code}: ErrorDetailsType = this._requestFilterError
-        ? {category: Error.Category.NETWORK, code: Error.Code.REQUEST_FILTER_ERROR}
-        : HlsJsErrorMap[errorName] || {category: 0, code: 0};
+      const {category, code}: ErrorDetailsType =
+        this._requestFilterError || this._responseFilterError
+          ? {category: Error.Category.NETWORK, code: this._requestFilterError ? Error.Code.REQUEST_FILTER_ERROR : Error.Code.RESPONSE_FILTER_ERROR}
+          : HlsJsErrorMap[errorName] || {category: 0, code: 0};
       HlsAdapter._logger.warn(new Error(Error.Severity.RECOVERABLE, category, code, errorDataObject));
     }
-    this._requestFilterError = false;
+    this._requestFilterError && (this._requestFilterError = false);
+    this._responseFilterError && (this._responseFilterError = false);
```

Comment:
isn't this just same as:
this._requestFilterError = false
this._responseFilterError = false
Why do we need to check if it is true before?